### PR TITLE
Bone meal tablets

### PR DESCRIPTION
**Pull request by FulcrumA**
#### Summary
SUMMARY: Content "Added bone meal tablets"

#### Purpose of change
As per discussion on the Discord from a couple of days ago, introducing bone meal tablets - a homemade calcium supplement.

#### Describe the solution
Aforementioned item was added, together with relevant recipe.

Whether such a pill would work and what would be its qualities was discussed. The math is as such:

https://www.feedipedia.org/node/222
>The ash content of the meat and bone meal normally ranges from 28 to 36 %; calcium is 7 to 10 % and phosphorus 4.5 to 6 %.

It's meat and bone but given the bone is the one with most of the calcium, just bone meal would be probably cloer (if not above) 10%. so 10mg per 100mg.
Then there's this:
https://www.urmc.rochester.edu/encyclopedia/content.aspx?contenttypeid=19&contentid=BoneMeal

>The RDA for calcium for adults under 50 years is 1,000 mg per day.

In game, a portion of bone meal is 0.45 kg = 450,000mg
10% of 450,000 is 45,000. That's 45x 100% of daily RDA of calcium. The recipe creates 40 100% calcium RDA pills however, to account for potential loss of material while working with powders (and for some balance as a secondary consideration) though may be adjusted (for example, @Regilith suggested 45 - 50 pills).

On the basis of https://www.dsld.nlm.nih.gov/dsld/dailyvalue.jsp informing us that the RDA requirement is raw amount of the nutrient ingested, not necessarily absorbed (with requirement befitting 97.5% of population as per http://www.nationalacademies.org/hmd/Global/News%20Announcements/How-the-RDA-for-Vitamin-D-Was-Determined.aspx) each pill provides aforementioned 100% of daily calcium intake.

While each pill won't be 100% calcium, I believe that simplifying it here won't affect the balance. We could account for mass of the pill being composed of other material and lower percentage of the calcium content in a pill itself, increasing number of pills produced to account for the total mass of bone meal but in game terms it will simply force the player to repeat pillow swallowing action several times with no other effect beside stacking joy penalties.

#### Additional context
I am concerned about the effect on balance. While I simulate several factors like how hard would be to ingest such pills (test performed with self-made calcium supplement, however created on the basis of eggshells), the PR severely decreases calcium deficiency being a concern, pushing it much further into realm of realism where indeed such deficiency is not really a threat for average person living in western society beside old or otherwise impacted ability to absorb and utilize calcium people.

Comments are welcome for the sake of discussing this balance effect as well as above decisions behind item values.

### CODE REVIEW

**Comment by nexusmrsep** [comment]
How do you remove the ash content etc.to end up with a 100% pure solution? Is it even achievable by home methods? If it's not achievable why not just create impure tablet with much less calcium content, and by doing this also achieve balance you seek.

**Comment by FulcrumA** [comment]
@nexusmrsep 
>If it's not achievable why not just create impure tablet with much less calcium content, and by doing this also achieve balance you seek.

>While each pill won't be 100% calcium, I believe that simplifying it here won't affect the balance. We could account for mass of the pill being composed of other material and lower percentage of the calcium content in a pill itself, increasing number of pills produced to account for the total mass of bone meal but in game terms it will simply force the player to repeat pillow swallowing action several times with no other effect beside stacking joy penalties.

Thus I worry it won't really change much in terms of balance. Yes, a lot of ash wouldn't be sieved through (though there also probably would be less ash in bone meal as compared to meat and bone meal) but then taking that into consideration wouldn't realistically lead merely to decrease of calcium content but also proportional increase of number of pills created and we'll be back where we are, just with forcing player to swallow several pills instead of one.

**Comment by Maddremor** [comment]
Are we making a distinction between calcium and bioavailable calcium?

**Comment by FulcrumA** [comment]
@Maddremor 
If I understand correctly, bioavailability refers to percentage of calcium absorbed by the body. If so, it's not an issue as the values are based on RDA requirement of particular quantity of calcium in miligrams that has to be ingested rather than absorbed.

**Comment by Maddremor** [comment]
Modern calcium supplements seem to come in the form of calium carbonate which is bioavailable. I looked at innawoods calcium sources about a year ago, and I didn't find any clear sources on calcium of bonemeal being bioavailable from digestion. Perhaps I didn't look hard enough, but some extra research might be prudent (looking into it now).

**Comment by Maddremor** [comment]
[This](https://academic.oup.com/ps/article/95/1/70/2461327) might be of interest.

**Comment by mlangsdorf** [comment]
Commercially available calcium supplements are about 40% calcium by mass and the rest is binders and and stabilizers.  I'd be amazed if a survivor could do better.

Also, a 1g pill of calcium is not all that large and shouldn't be hard to swallow - they're chewable.  I can believe they'd taste terrible - at best they're going to taste like chalk, and these sound like they'd taste like chalk and dirt.  More skilled survivors might try to make flavored tablets by adding dried fruit to the mix.  Berry flavored chalk isn't much better than chalk but it does help.

If calcium supplements are going to be readily available, it might be worthwhile to add a hypercalcemia condition, though it's pretty hard for normal humans to go hypercalcemic since the body does regulate the absorption of calcium.  That's a general issue with CDDA's vitamin model.

**Comment by Maddremor** [comment]
After some searching I found that bone in calcium comes in the form of [Hydroxyapatite](https://en.wikipedia.org/wiki/Hydroxyapatite). I haven't found how absorbable that is during digestion or how bone derived supplements were manufactured. It's possible that some kind of leeching with acid is used to break the bone structure down, but that's just a guess on my part.

File: data/json/items/comestibles/med.json [hunk 1]
```diff
@@ -1167,6 +1167,29 @@
...
             ]
         }
+    },
+        {
+        "id" : "bonemeal_tablet",
+        "type" : "COMESTIBLE",
+        "comestible_type" : "MED",
+        "name" : "bone meal tablet",
+        "description" : "Homemade calcium supplement made out of bone meal. Tastes horribly and is hard to swallow but it does its job.",
+        "weight" : 1,
```
Comment (mlangsdorf) [inline data/json/items/comestibles/med.json]: Make the weight 2g per pill and then they've got a realistic 50% calcium content.

File: data/json/items/comestibles/med.json [hunk 2]
```diff
@@ -1167,6 +1167,29 @@
...
+        "volume" : 1,
+        "price" : 10,
+        "fun" : -8,
+        "price_postapoc": 1000,
+        "charges" : 40,
+        "stack_size" : 40,
+        "symbol" : "!",
+        "color" : "white",
```
Comment (mlangsdorf) [inline data/json/items/comestibles/med.json]: Add a looks_like so they have a tile in tilesets that don't support them yet.

File: data/json/items/comestibles/med.json [hunk 3]
```diff
@@ -1167,6 +1167,29 @@
...
+        {
+        "id" : "bonemeal_tablet",
+        "type" : "COMESTIBLE",
+        "comestible_type" : "MED",
+        "name" : "bone meal tablet",
+        "description" : "Homemade calcium supplement made out of bone meal. Tastes horribly and is hard to swallow but it does its job.",
+        "weight" : 1,
+        "volume" : 1,
+        "price" : 10,
+        "fun" : -8,
```
Comment (mlangsdorf) [inline data/json/items/comestibles/med.json]: This fun is reasonable for an ash and bone pill, but I think a flavored bonemeal tablet with fun: -1 or so is also reasonable.

File: data/json/recipes/recipe_medsandchemicals.json [hunk 4]
```diff
@@ -242,6 +242,27 @@
...
+},{
+  "type" : "recipe",
+  "result": "bonemeal_tablet",
+  "category": "CC_CHEM",
+  "subcategory": "CSC_CHEM_DRUGS",
+  "skill_used": "cooking",
+  "skills_required": [ "survival", 2 ],
+  "difficulty": 2,
+  "time": 20000,
+  "book_learn": [["pocket_survival", 3] , ["atomic_survival", 2] , ["textbook_survival", 2] , ["survival_book", 3]],
```
Comment (mlangsdorf) [inline data/json/recipes/recipe_medsandchemicals.json]: Recipe looks fine.  Increase the difficulty to 3 and add dried fruit or fruit leathers for the flavored version.

**Comment by FulcrumA** [comment]
Some changes introduced as per @mlangsdorf's review. Do tell if it works fine, as I may have missed something, especially since I am not sure if you wanted to decrease RDA calcium per tablet or just merely increase one's size. Also, it's not merely size that make such homemade supplement hard to swallow - it's the texture, at least for me.

File: data/json/items/comestibles/med.json [hunk 5]
```diff
@@ -1167,6 +1167,54 @@
...
+        "charges" : 40,
+        "stack_size" : 40,
+        "symbol" : "!",
+        "color" : "white",
+        "looks_like" : "calcium_tablet",
+        "use_action" : {
+            "type" : "consume_drug",
+            "activation_message" : "You take the %s.",
+            "vitamins": [
+                [ "calcium", 12 ]
```
[the remark this answers is not in the document]
Comment (FulcrumA) [reply data/json/items/comestibles/med.json]: Ah, alright, that's what I was wondering about. Will adjust it back to what it was, with similar value for the flavored variant.

**Comment by dissociativity** [comment]
So, this should give vitamin (calcium) but be unhealthy? (lower health stat a bit)
That seems fair.

**Comment by FulcrumA** [comment]
@dissociativity 
Some time I've spent to mull it over and I think it's be a pretty good idea. Even post-apoc animals "freegrazing" still likely would suffer some token toxic metal buildup in their bones (especially given that environment in many areas seems pretty polluted) and on top of that it'd make the tablets much more consistent with other medication consumables which much of seems to be somewhat unhealthy.

I think that for now, since it was quite a bit since PR's creation and the last comment and so between them and the review we have the bone meal supplements balanced relatively well for what role they are to serve, while still realistic - it will be the last change, after which I'll change the tag/name to indicate this PR is done.

**Comment by FulcrumA** [comment]
For the sake of reference - it should partially alleviate the issue in #18625 together with other vitamin/nutrient work that is currently going on (albeit depending on one's scenario, certain nutrients should be harder to acquire).

**Pull request merged**